obspy/io/nied/fnetmt/core.py

```diff
@@ -0,0 +1,258 @@
+
+
+class FNETMTException(Exception):
+    pass
+
+
+def _get_resource_id(name, res_type, tag=None):
+    """
+    Helper function to create consistent resource ids.
+    """
+    res_id = "smi:local/fnetmt/%s/%s" % (name, res_type)
+    if tag is not None:
+        res_id += "#" + tag
+    return res_id
+
+
+def _buffer_proxy(filename_or_buf, function, reset_fp=True,
+                  file_mode="rb", *args, **kwargs):
+    """
+    Calls a function with an open file or file-like object as the first
+    argument. If the file originally was a filename, the file will be
+    opened, otherwise it will just be passed to the underlying function.
+
+    :param filename_or_buf: File to pass.
+    :type filename_or_buf: str, open file, or file-like object.
+    :param function: The function to call.
+    :param reset_fp: If True, the file pointer will be set to the initial
+        position after the function has been called.
+    :type reset_fp: bool
+    :param file_mode: Mode to open file in if necessary.
+    """
+    try:
+        position = filename_or_buf.tell()
+        is_buffer = True
+    except AttributeError:
+        is_buffer = False
+
+    if is_buffer is True:
+        ret_val = function(filename_or_buf, *args, **kwargs)
+        if reset_fp:
+            filename_or_buf.seek(position, 0)
+        return ret_val
+    else:
+        with open(filename_or_buf, file_mode) as fh:
+            return function(fh, *args, **kwargs)
+
+
+def _is_fnetmt_catalog(filename_or_buf):
+    """
+    Checks if the file is a F-net moment tensor file.
+
+    :param filename_or_buf: File to test.
+    :type filename_or_buf: str or file-like object.
+    """
+    try:
+        return _buffer_proxy(filename_or_buf, __is_fnetmt_catalog,
+                             reset_fp=True)
+    # Happens for example when passing the data as a string which would be
+    # interpreted as a filename.
+    except (OSError):
+        return False
+
+
+def __is_fnetmt_catalog(buf):
+    """
+    Test whether file is an F-net moment tensor catalog file by reading the
+    header and the first data line. Reads at most 40 lines.
+
+    :param buf: File to read.
+    :type buf: Open file or open file like object.
+    """
+    cnt = 0
+    try:
+        while True:
+            line = buf.readline()
+            if not line:
+                return False
+            line = line.decode()
+            # read at most 40 lines
+            if cnt > 40:
+                return False
+            if line.find('Total Number') != -1:
+                match = re.search(r'Total Number:\s+(\d+)\s+', line)
+                if match:
+                    nevents = int(match.group(1))
+            if line.startswith('Origin Time'):
+                if nevents > 0:
+                    data = buf.readline()
+                    a = data.split()
+                    if len(a) != 21:
+                        return False
+                    return True
+            cnt += 1
+    except:
```

Comment:
What kind of exceptions are you expecting here? The only one I can see is the `int` conversion.